addons/calendar_bis/models/calendar_timeslot.py

```diff
@@ -0,0 +1,195 @@
+from odoo import models, fields, api, _, Command
+from odoo.exceptions import AccessError
+
+from dateutil.relativedelta import relativedelta
+
+days = ['mo', 'tu', 'we', 'th', 'fr', 'sa', 'su']
+
+class CalendarTimeslot(models.Model):
+    _name = "calendar.timeslot"
+    _description = "Calendar Timeslot"
+    _order = "start asc, id"
+
+    # Technical Fields
+    event_id = fields.Many2one('calendar.event', default=lambda self: self.env['calendar.event'].create([{}]).id)
+    can_read_private = fields.Boolean(compute='_compute_access', default=True)
+    can_write = fields.Boolean(compute='_compute_access', default=True)
+    active = fields.Boolean(default=True)
+    edit = fields.Selection([('one', 'This event only'),
+                             ('all', 'All events in the series'),
+                             ('post', 'All event after this one')], default='all', store=False)
+
+    # Time Related Fields
+    start = fields.Datetime(default=fields.Datetime.now, required=True)
+    stop = fields.Datetime(required=True, compute='_compute_stop', readonly=False, store=True)
```

Comment:
In the current implementation, we have start, stop and start_date and stop_date it allows to display all_day events. I see that you removed these fields. In calendar we have two complex cases to handle:

1. Jerémy needs to join a meeting online at 17:00. But the organizer, Alan is in another timezone. The system will display 17:30 to Jeremy and it will display 12:30 to Alan. I think your implementation should work out of the box for that case. 
2. The other case which is more difficult is the event common for several people but occurring at the same date. Alain and Jeremy will be off for Christmas day but it should display the whole day for both of them.
3. Moreover, their company is giving them the 24th of December off. It should be displayed for all of them at the same time. 

The second situation is quite difficult to handle. We somewhat support it with the all day events that have a forced date and the _inverse_dates method that will force the datetime of all day events from 8 to 18. It works only if you are not too far in Japan or extreme Russia. How to you plan to handle these kind of things?

Reply:
Yes, All day is not implemented yet but will be in the final version